lib/ecto/model/callbacks.ex

```diff
@@ -109,12 +109,17 @@ defmodule Ecto.Model.Callbacks do
       before_insert User, :generate_permalink
 
   """
-  defmacro before_insert(function, args \\ []),
-    do: register_callback(:before_insert, function, args, [])
+  defmacro before_insert(function),
+    do: register_callback(:before_insert, function, [])
+
+  defmacro before_insert(function, args)
+    when is_list(args),
+    do: register_callback(:before_insert, function, args)
+
+  defmacro before_insert(module, function)
+    when is_atom(function),
+    do: register_callback(:before_insert, module, function, [])
 
-  @doc """
```

Comment:
Don't remove the docs. It is another function and we need to document it as such. That's the main issue with this PR, it adds yet another function to document! :)